components/list/list-item-mixin.js

```diff
@@ -595,6 +624,11 @@ export const ListItemMixin = superclass => class extends composeMixins(
 		const alignNested = ((this.draggable && this.selectable) || (this.expandable && this.selectable && this.color)) ? 'control' : undefined;
 		const primaryAction = ((!this.noPrimaryAction && this._renderPrimaryAction) ? this._renderPrimaryAction(this._contentId) : null);
 		const tooltipForId = (primaryAction ? this._primaryActionId : (this.selectable ? this._checkboxId : null));
+
+		const rootList = this.getRootList();
```

Comment:
A couple of different thoughts here...

Is it possible to have a scenario where a nested list has the add button, but the ancestor list does not? Just wondering whether we should be checking the list that the item belongs to vs the root list.

Ideally if the list's `add-button` property changes, the list-items update themselves to show/hide the add-buttons as needed. It kinda seems like the items would add/remove the add-button as a result of the item's `render` method running for other reasons. Is that correct? Do we need something more to handle this case?

Kinda related to the above, I'm curious how often this runs, and whether we can avoid calling `getRootList` every time the item's `render` is called.  I'm not certain but I suspect that the render method may get called often, as the user interacts (hovers, focuses, selects, expands, drags, etc).

Reply:
Great points
1 - I'll confirm with Jeff but my understanding is that we want this to be set at the root list level and would not want consumers just enabling `add-button` for certain nested lists
2 - You are correct on how it should behave and how it currently is. I'll look into if there's a better way to handle this so if `add-button` changes then list items update
3 - I also think it runs fairly often. I'll see if there's a better way to avoid calling `getRootList` so often.

Reply:
Checked with Jeff - we actually want this to just be a property on each list (similar to `grid`) so nested lists would have to opt-in as well. I think that drastically simplifies all of this.